File: SYSMCLTD.Customers.Data/Models/Customer.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace SYSMCLTD.Customers.Data.Models
{
    public partial class Customer
    {
        public Customer()
        {
            Addresses = new HashSet<Address>();
            Contacts = new HashSet<Contact>();
        }

        public int Id { get; set; }
        public bool IsDeleted { get; set; }
        //public DateTime Created { get; set; }
        public string Name { get; set; }
        public int CustomerNumber { get; set; }

        public virtual ICollection<Address> Addresses { get; set; }
        public virtual ICollection<Contact> Contacts { get; set; }
    }
}

File: SYSMCLTD.Customers.Client/src/app/models/customer.ts
export class BaseModel{
    id: number;
    isDeleted: boolean;
    constructor(id: number, isDeleted: boolean){
        this.id = id;
        this.isDeleted = isDeleted;
      }
} 


export class Address extends BaseModel{ 
  city: string;
  street: string;
  customerId: number;
  constructor(id: number, isDeleted: boolean, city: string, street: string, customerId:number){
    super(id,isDeleted);
    this.city = city;
    this.street = street;
    this.customerId= customerId;
  }
}

export class Contact extends BaseModel {
    fullName: string;
    officeNumber?: number;
    email?: string;
    customerId:number;

    constructor(id: number, isDeleted: boolean, fullName: string, customerId:number,officeNumber?: number, email?:string){
      super(id,isDeleted);
      this.fullName = fullName;
      this.officeNumber = officeNumber;
      this.email= email;
      this.customerId = customerId;
    }
}

export class Customer extends BaseModel{
    name:string;
    customerNumber: number;
    addresses?:Array<Address>;
    contacts?: Array<Contact>;

    constructor(id: number, isDeleted: boolean, name: string,customerNumber: number){
        super(id,isDeleted);
        this.name = name;
        this.customerNumber = customerNumber;
    }
  }